service/builder/exporters_builder.go

```diff
@@ -29,24 +29,15 @@ import (
 // builtExporter is an exporter that is built based on a config. It can have
 // a trace and/or a metrics consumer and have a shutdown function.
 type builtExporter struct {
-	logger *zap.Logger
-	te     component.TraceExporter
-	me     component.MetricsExporter
-	le     component.LogsExporter
+	logger    *zap.Logger
+	expByType map[configmodels.DataType]component.Exporter
 }
 
 // Start the exporter.
-func (exp *builtExporter) Start(ctx context.Context, host component.Host) error {
+func (bexp *builtExporter) Start(ctx context.Context, host component.Host) error {
 	var errors []error
-	if exp.te != nil {
-		err := exp.te.Start(ctx, host)
-		if err != nil {
-			errors = append(errors, err)
-		}
-	}
-
-	if exp.me != nil {
-		err := exp.me.Start(ctx, host)
+	for _, exporter := range bexp.expByType {
```

Comment:
The assumption here is that a built exporter only has entries for the created data types, I mean `otlp` will only have a "metrics" entry if one was actually created for that exporter.

Reply:
I saw the assumption confirmed below.